### PR TITLE
*: Enforce that the AllNamespace install mode has been enabled

### PR DESCRIPTION
**Pull request by timflannagan**
Signed-off-by: timflannagan <timflannagan@gmail.com>

### CODE REVIEW

File: testdata/bundles/registry/invalid/manifests/leader-election-role_rbac.authorization.k8s.io_v1_role.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (joelanford) [inline testdata/bundles/registry/invalid/manifests/leader-election-role_rbac.authorization.k8s.io_v1_role.yaml:1]: Seems like a bunch of these manifests are irrelevant for the test of "AllNamespaces or not".

Can we get rid of everything except the CSV?

Comment (timflannagan) [reply testdata/bundles/registry/invalid/manifests/leader-election-role_rbac.authorization.k8s.io_v1_role.yaml:1]: Yup, good catch. That fine with me.

Comment (timflannagan) [reply testdata/bundles/registry/invalid/manifests/leader-election-role_rbac.authorization.k8s.io_v1_role.yaml]: Done :+1:

**Comment by timflannagan** [comment]
I can also try to strip down the CSV to just the spec.InstallModes.

**Review by exdx — APPROVED** [review]

File: internal/convert/registryv1.go [hunk 2]
```diff
@@ -190,6 +190,9 @@ func Convert(in RegistryV1, installNamespace string, targetNamespaces []string)
 			supportedInstallModes.Insert(string(im.Type))
 		}
 	}
+	if !supportedInstallModes.Has(string(v1alpha1.InstallModeTypeAllNamespaces)) {
```
[the remark this answers is not in the document]
Comment (timflannagan) [reply internal/convert/registryv1.go:193]: I think we're all set if this didn't fail on the "valid" registry+v1 test case?

**Review by joelanford — APPROVED** [review]

**Pull request merged**